engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java

```diff
@@ -46,7 +46,16 @@ public static String findExceptionMessage(Throwable ex) {
   public static boolean isInterruptedException(Throwable ex) {
     var iop = InteropLibrary.getUncached();
     try {
-      return iop.getExceptionType(ex) == ExceptionType.INTERRUPT;
+      var interrupt = iop.getExceptionType(ex) == ExceptionType.INTERRUPT;
+      if (interrupt) {
+        return true;
+      }
+      try {
+        var cause = iop.getExceptionCause(ex);
+        return cause != null && iop.getExceptionType(cause) == ExceptionType.INTERRUPT;
```

Comment:
This code check `ex` and `ex.getCause()`. What if the chain is deeper. Shouldn't we check all `ex.getCause().getCause().getCause()....`?

Reply:
I've been wondering if it makes sense to handle multiple levels. But I guess one could go crazy and propagate it that way. It probably doesn't hurt to add a logic for that.